_rules/text-spacing-not-important-b8bb68.md

```diff
@@ -0,0 +1,288 @@
+---
+id: b8bb68
+name: Text spacing in `style` attributes is not `!important`
+rule_type: atomic
+description: |
+  This rule checks that text spacing set in the `style` attribute is below permissible minimum value or not set to `!important`
+accessibility_requirements:
+  wcag21:1.4.12: # Text Spacing (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Jey Nandakumar
+---
+
```

Comment:
I think the rule would be punchier by checking for the !important flag in the applicability. The expectation only has to check that the computed value is within acceptable range. That would move part of the duplication in expectations to the applicability and remove the duplication.

That would make many of the passed examples inapplicable instead, but that is OK.

Reply:
+1 ... that seems like a reasonable alternative approach.

Reply:
Actually, looking more closely, I'm not sure "!important" needs to be in the applicability. I think a some other adjustments might help, and maybe splitting into 3 rules if lots of test cases are necessary. If I understand the SC correctly, then something like this perhaps...

**Description** 
This rule checks that the text spacing styles are able to be adjusted, ie. `!important` is not applied to the style, unless the style has already exceeded the specified metric.

**Applicability** 
This rule applies to any HTML element that is visible and has one of the following CSS properties set by the style attribute:
- line-height
- word-spacing
- letter-spacing

**Expectation 1** (for example)
The test target does not have the important flag set for the declared value of the CSS line-height property that is specified by the style attribute of the test target, **or** the computed line-height is at least 1.5 times the computed font-size of the test target.

**Passed Example 1** (for example)
This p element has a line-height less than 1.5 times the font size and does not have the important flag applied, so a user may increase the line-height above the threshold should they wish to.

```
<html>
	<style>
		body { font-size: 1em; }
	</style>
	<body>
		<p style="line-height: 1.25em;">
			I was in darkness, but I took three steps and found myself in paradise. The first step was a good thought, the second, a good word; and the third, a good deed.
		</p>
	</body>
</html>
```

**Passed Example** 2
This p element has a line-height greater than 1.5 times the font size so it is not a requirement for a user to be able to adjust the line-height.

```
<html>
	<style>
		body { font-size: 1em; }
	</style>
	<body>
		<p style="line-height: 1.75em; !important">
			I was in darkness, but I took three steps and found myself in paradise. The first step was a good thought, the second, a good word; and the third, a good deed.
		</p>
	</body>
</html>
```

**Failed Example** 1
This p element has a line-height less than 1.5 times the font size and does have the important flag applied, so a user would be unable to increase the line-height above the threshold should they wish to.

```
<html>
	<style>
		body { font-size: 1em; }
	</style>
	<body>
		<p style="line-height: 1.25em; !important">
			I was in darkness, but I took three steps and found myself in paradise. The first step was a good thought, the second, a good word; and the third, a good deed.
		</p>
	</body>
</html>
```